cfgen/cfgen

```diff
@@ -1423,6 +1440,58 @@ def cluster_svcs(m0conf: Dict[Oid, Any], svc_t) -> List[Oid]:
             if m0conf[svc_id].type is svc_t]
 
 
+# node-to-drives mapping
+def pool_drives_with_nodes(m0conf: Dict[Oid, Any]) -> List[Tuple[Oid, Oid]]:
+    all_drives: List[Tuple[Oid, Oid]] = [
+        (drive_id, m0conf[encl_id].node)
+        for encl_id in m0conf if encl_id.type is ObjT.enclosure
+        for ctrl_id in m0conf[encl_id].ctrls
+        for drive_id in m0conf[ctrl_id].drives
+    ]
+    assert all_drives
+    assert all(drive_id.type is ObjT.drive and node_id.type is ObjT.node
+               for drive_id, node_id in all_drives)
+
+    return all_drives
+
+
+def pool_drives_from_pooltypes(
+                cluster_desc: Dict[str, Any],
+                pool_desc: Dict[str, Any]):
+
+    for pool in cluster_desc['pools']:
+        ptype = pool_type(pool_desc)
+        if ptype is PoolT.sns:
+            disk_refs = get_pool_diskrefs(pool)
+        if ptype is PoolT.dix:
+            disk_refs = get_pool_diskrefs(pool)
+
+
+def pool_drives_from_references(
+                m0conf: Dict[Oid, Any],
+                pool_desc: Dict[str, Any],
+                cluster_desc: Dict[str, Any]) -> List[Tuple[Oid, Oid]]:
+    disk_refs = pool_drives_from_pooltypes(cluster_desc, pool_desc)
+    drives = []
+    all_drives: List[Tuple[Oid, Oid]] = pool_drives_with_nodes(m0conf)
+
+    if disk_refs:
```

Comment:
You need this additional if condition, for loop will terminate if disk_refs list is empty.

Reply:
If I remove this if condition here, then it throws error like:
--> Checking cfgen cfgen:1481: error: Item "None" of "Optional[Any]" has no attribute "__iter__" (not iterable)